operations/configuration_management/NCN_Worker_Image_Customization.md

```diff
@@ -0,0 +1,235 @@
+
+**NOTE:** Some of the documentation linked from this page mentions use of the Boot Orchestration Service (BOS). The use of BOS
+is only relevant for booting compute nodes and can be ignored when working with NCN images.
+
+This document describes the configuration of a Kubernetes NCN image. The same steps could be used to modify
+a Ceph image.
+
+1. Locate the NCN Image to be Modified
+
+    This example assumes the administrator wants to modify the Kubernetes image that is currently in use by NCNs. However, the steps are the same for any NCN SquashFS image.
+
+    ```console
+    ncn# ARTIFACT_VERSION=<your-version>
+
+    ncn# cray artifacts get ncn-images k8s/$ARTIFACT_VERSION/filesystem.squashfs ./$ARTIFACT_VERSION-filesystem.squashfs
+
+    ncn# cray artifacts get ncn-images k8s/$ARTIFACT_VERSION/kernel ./$ARTIFACT_VERSION-kernel
+
+    ncn# cray artifacts get ncn-images k8s/$ARTIFACT_VERSION/initrd ./$ARTIFACT_VERSION-initrd
+
+    ncn# export IMS_ROOTFS_FILENAME=$ARTIFACT_VERSION-filesystem.squashfs
+
+    ncn# export IMS_KERNEL_FILENAME=$ARTIFACT_VERSION-kernel
+
+    ncn# export IMS_INITRD_FILENAME=$ARTIFACT_VERSION-initrd
+    ```
+
+1. [Import External Image to IMS](../image_management/Import_External_Image_to_IMS.md)
+
+    This document will instruct the administrator to set several environment variables, including the three set in
+    the previous step.
+
+1. Clone the `csm-config-management` repository
+
+   ```bash
+   git clone https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git
+   ```
+
+1. [Create a CFS Configuration](Create_a_CFS_Configuration.md)
+
+   **NOTE:** A new `initrd` must be generated at the end of the CFS session. Add the following Ansible
+   playbook to the `csm-config-management` repository :
+
+   `ncn-initrd.yml`:
+
+   ```console
+   - hosts: Management_Worker
+     any_errors_fatal: true
+     remote_user: root
+
+     tasks:
+     - name: Create NCN initrd
+       command: /srv/cray/scripts/common/create-ims-initrd.sh
+       when: cray_cfs_image|default(false)|bool
+   ```
+
+   This playbook should be the final layer in the CFS configuration you create in following steps.
+
+   **NOTE:** If desired, the default SSH *host* keys can be removed with a CFS Ansible task:
+
+   ```console
+   - name: Remove SSH host keys
+     command: rm -f /etc/ssh/*key*
+     warn: false
+   ```
+
+   New host keys will be generated by `sshd` when the NCN boots for the first time.
+
+1. [Create an Image Customization CFS Session](Create_an_Image_Customization_CFS_Session.md)
+
+   In this section, use the following values for the target definition and target group for the
+   `cray cfs session create` command invocation:
+
+   ```console
+   --target-definition image
+   --target-group Management_Worker
+   ```
+
+1. Download the Resultant NCN Artifacts
+
+    **NOTE:** `$IMS_RESULTANT_IMAGE_ID` is the `result_id` returned in the output of the last command
+    in the previous step:
+
+    ```console
```

Comment:
Is this the last command of the previous step, and then the next block of commands are the ones to be run in this step?